der/src/asn1/internal_macros.rs

```diff
@@ -1,17 +1,47 @@
-macro_rules! impl_string_type {
+macro_rules! impl_type {
```

Comment:
I splitted it as I think I can just reuse this in BitString, GeneralizedTime, Int*, OctetString and UtcTime